src/custom/hooks/useParseReferralQueryParam.ts

```diff
@@ -0,0 +1,26 @@
+import { useMemo } from 'react'
+import { isAddress } from 'ethers/lib/utils'
+import useParsedQueryString from 'hooks/useParsedQueryString'
+import { REFERRAL_QUERY_PARAM } from 'hooks/useReferralLink'
+
+/**
+ * Returns the parsed referral address from the query parameters if its a valid address
+ */
+export default function useParseReferralQueryParam(): string | null {
+  const parsedQs = useParsedQueryString()
+
+  const referral = useMemo(() => {
+    const referralAddress = parsedQs[REFERRAL_QUERY_PARAM]
+    if (typeof referralAddress === 'string' && isAddress(referralAddress)) {
```

Comment:
You can use `isAddress` from utils https://github.com/gnosis/cowswap/blob/b366fa833887f6a86dacf93f425e2dc6f250c811/src/utils/index.ts#L10

With that you don't need the extra string type check